fdbserver/workloads/ClientMetric.actor.cpp

```diff
@@ -0,0 +1,253 @@
+		state RangeResult txInfoEntries;
+		// wait to make sure client metrics are updated
+		wait(delay(CLIENT_KNOBS->CSI_STATUS_DELAY));
+		loop {
+			try {
+				tr->reset();
+				tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+				tr->setOption(FDBTransactionOptions::LOCK_AWARE);
+				std::string sampleRateStr = "default";
+				std::string sizeLimitStr = "default";
+				const double sampleRateDbl =
+				    cx->globalConfig->get<double>(fdbClientInfoTxnSampleRate, std::numeric_limits<double>::infinity());
+				if (!std::isinf(sampleRateDbl)) {
+					sampleRateStr = std::to_string(sampleRateDbl);
+				}
+				const int64_t sizeLimit = cx->globalConfig->get<int64_t>(fdbClientInfoTxnSizeLimit, -1);
+				if (sizeLimit != -1) {
+					sizeLimitStr = std::to_string(sizeLimit);
+				}
+				std::cout << "Read from globalconfig: sampleRate=" << sampleRateStr << " sizeLimit=" << sizeLimitStr
+				          << std::endl;
+				state RangeResult kvRange = wait(
+				    tr->getRange(begin, end, keysLimit, Snapshot::False, reverse ? Reverse::True : Reverse::False));
+				if (kvRange.empty()) {
+					wait(delay(1.0));
+					std::cout << "WaitingForLatencyMetricToBePresent" << std::endl;
+					TraceEvent("WaitingForLatencyMetricToBePresent").log();
+					continue;
+				}
+				txInfoEntries.arena().dependsOn(kvRange.arena());
+				txInfoEntries.append(txInfoEntries.arena(), kvRange.begin(), kvRange.size());
+				break;
+			} catch (Error& e) {
+				wait(tr->onError(e));
+			}
+		}
+		for (auto& kv : txInfoEntries) {
+			uint64_t vs = getVersionStamp(kv.key);
+			std::cout << "VersionStamp is " << vs << std::endl;
+		}
+		return txInfoEntries;
+	}
+
+	ACTOR Future<Void> writeRandomKeys(Database cx, int total) {
+		state int cnt = 0;
+		state Transaction tr(cx);
+		try {
+			loop {
+				try {
+					wait(delay(0.001));
+					tr.reset();
+					tr.set(Key(deterministicRandom()->randomAlphaNumeric(10)),
+					       Value(Key(deterministicRandom()->randomAlphaNumeric(10))));
+					wait(tr.commit());
+					if (cnt >= total) {
+						break;
+					}
+					++cnt;
+				} catch (Error& e) {
+					wait(tr.onError(e));
+				}
+			}
+		} catch (Error& e) {
+			TraceEvent(SevError, "ClientMetricErrorWhenWriteKeys").error(e);
+			throw;
+		}
+		std::cerr << "writeRandomKeys finish, written=" << cnt << std::endl;
+		return Void();
+	}
+
+	// goal:
+	//      write some random keys, check the latency metric and the latest version stamp vs1
+	//      write some other random keys, check the latency metric and latest version stamp again vs2
+	//      vs2 should be strictly larger than vs1, to verify new latency metrics are added
+	ACTOR Future<Void> runner(Database cx, ClientMetricWorkload* self) {
+		try {
+			state int initialWrites = deterministicRandom()->randomInt(1, 5);
+			state int secondWrites = deterministicRandom()->randomInt(1, 5);
+			state int keysLimit = 1;
+
+			state int retry1 = 0;
+			state int max_retry1 = 10;
+			loop {
```

Comment:
This loop and the loop below are almost identical. Please refactor the code into function calls.